### PR TITLE
EPMLSTRCMW-222

### PR DESCRIPTION
**Pull request by OlegKoshelev**
**Fix migrate to codecov.**

- cmwl_pipeline_ci.yml,  build.sbt and project/plugins.sbt  have been updated.
-  codecov.yml has been added.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -28,7 +28,7 @@ lazy val commonSettings = Seq(
   testOptions in Test += Tests.Argument(TestFrameworks.ScalaTest, "-h", "target/test-reports"),
   coverageEnabled in Test := true,
   coverageMinimum := 60,
-  coverageFailOnMinimum := true,
+  coverageFailOnMinimum := false,
```
Comment (OlegKoshelev) [inline build.sbt]: @ilyaonishenko Now seems it works but I'm not sure about this line. If use "true"  the tests won't pass due to coverage  <  60% ( you can see it during run test job). Maybe you can suggest a better solution?

**Review by Nikolai-Isaev — APPROVED** [review]
It works, nice job! Will this bot always comment on coverage here btw?

**Review by DruzhininPavel — APPROVED** [review]
Yep. it is normal!

**Comment by OlegKoshelev** [comment]
> It works, nice job! Will this bot always comment on coverage here btw?
i think so yes

**Review by gominasian — APPROVED** [review]

File: build.sbt [hunk 2]
```diff
@@ -27,9 +27,10 @@ lazy val commonSettings = Seq(
   test in Test := (test in Test).dependsOn(checkFormat).value,
   testOptions in Test += Tests.Argument(TestFrameworks.ScalaTest, "-h", "target/test-reports"),
   coverageEnabled in Test := true,
-  coverageMinimum := 60,
+  coverageMinimum := 50,
```
Comment (iliaonishchenko) [inline build.sbt:30]: Does it work with coverage minimum = 60?

File: codecov.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+coverage:
+  status:
+    project:
+      default:
+        target: auto
+        treshold: 0%
```
Comment (iliaonishchenko) [inline codecov.yml]: What threshold stands for?

**Review by iliaonishchenko — COMMENTED** [review]
Nice work, I have some questions:

P.S. also please fix commit message. 'Migrate to codecov[2]' sounds better then 'Fix migrate to codecov'

File: build.sbt (hunk 2 above)
Comment (OlegKoshelev) [reply build.sbt:30]: Thanks for the clarification! It seems possible. I'll check, but if it doesn't work, I'll leave 50

File: codecov.yml (hunk 3 above)
Comment (OlegKoshelev) [reply codecov.yml]: As I understood it allows to monitor coverage. If next MR have coverage less it won't pass. For example if threshold = 10% and current coverage 70%, it allows to reduce coverage at next MR to 60%. I think I made a typo, I'll fix it.

File: build.sbt (hunk 2 above)
Comment (OlegKoshelev) [reply build.sbt:30]: @ilyaonishenko Seems it doesn't work with 60% 
(Coverage is below minimum [51.65% < 60.00%]: Stmt:Total
Error:  (womtool / coverageReport) Coverage minimum was not reached)

**Review by GrigoriiBerezin — APPROVED** [review]
Nice

**Pull request merged**